File: Solutions/0028/0028.py
# -*- coding:utf-8 -*-
# Author: washing
# DateTime: 2020/12/28 17:58
# File: 0028.py
# Desc: 

class Solution:
    def strStr(self, haystack: str, needle: str) -> int:
        # return haystack.find(needle)
        ret = -1
        for i in range(len(haystack)-len(needle)+1):
            if needle == haystack[i:i+len(needle)]: return i
        return ret

File: Solutions/0028/0028.java
// Author: washing
// DateTime: 2021/4/20 10:08
// File: 0028.java
// Desc: 

class Solution {
    public int strStr(String haystack, String needle) {
        int l = needle.length();
        if (needle.equals("")) return 0;
        for(int i=0; i<=haystack.length()-l; i++){
            if(haystack.substring(i, i+l).equals(needle)) return i;
        }
        return -1;
    }
}